src/python/pants/core/target_types_test.py

```diff
@@ -367,3 +372,125 @@ def gen_resource_tgt(rel_fp: str, tags: list[str] | None = None) -> ResourceTarg
         gen_resource_tgt("f2.ext"),
         gen_resource_tgt("subdir/f.ext"),
     }
+
+
+@pytest.mark.parametrize("asset_type", ("file", "resource"))
+def test_url_assets(asset_type) -> None:
+    rule_runner = RuleRunner(
+        rules=[
+            *target_type_rules(),
+            *pex_from_targets.rules(),
+            *package_pex_binary.rules(),
+            *run_pex_binary.rules(),
+            *python_target_type_rules.rules(),
+            *run.rules(),
+        ],
+        target_types=[FileTarget, ResourceTarget, PythonSourceTarget, PexBinary],
+        objects={"http_source": HTTPSource},
+    )
+    http_source_info = (
+        'url="https://raw.githubusercontent.com/python/cpython/7e46ae33bd522cf8331052c3c8835f9366599d8d/Lib/antigravity.py",'
+        "len=500,"
+        'sha256="8a5ee63e1b79ba2733e7ff4290b6eefea60e7f3a1ccb6bb519535aaf92b44967"'
+    )
+    rule_runner.write_files(
+        {
+            "assets/BUILD": dedent(
+                f"""\
+                {asset_type}(
+                    name='antigravity',
+                    source=http_source(
+                        {http_source_info},
+                    ),
+                )
```

Comment:
Ooh. I like this.

Thanks a lot!

Reply:
I'm really glad I didn't stop at `url=` and got this working with `source`.

I'm also really excited to see how people use this. I already know how in my own repo because I have this in a plugin :see_no_evil:

Reply:
I think that would be super cool

Reply:
https://github.com/pantsbuild/pants/issues/15503